relays/lib-substrate-relay/src/messages_source.rs

```diff
@@ -126,16 +129,19 @@ where
 		&self,
 		id: SourceHeaderIdOf<MessageLaneAdapter<P>>,
 	) -> Result<(SourceHeaderIdOf<MessageLaneAdapter<P>>, MessageNonce), SubstrateError> {
-		let encoded_response = self
+		let outbound_lane_data: Option<OutboundLaneData> = self
 			.client
-			.state_call(
-				P::TargetChain::TO_CHAIN_LATEST_GENERATED_NONCE_METHOD.into(),
-				Bytes(self.lane_id.encode()),
+			.storage_value(
+				outbound_lane_data_key(
+					P::TargetChain::WITH_CHAIN_MESSAGES_PALLET_NAME,
+					&self.lane_id,
+				),
```

Comment:
where is the code that _puts_ this data to storage?

Reply:
We can only modify storage through transaction, so it's in the messages module itself (https://github.com/paritytech/parity-bridges-common/blob/master/modules/messages/src/lib.rs#L1062-L1081). Relay, otoh, has two options - one to use runtime API and query runtime storage directly. So this PR replaces 1st option with 2nd, because it is easier to maintain.

Reply:
ok, makes sense